### PR TITLE
RSDK-4369 - Create jammy base and rebase cpp-sdk

### PR DESCRIPTION
**Pull request by seanavery**
## Description
- Creates new `jammy` base image for building the appimage.
- Adjusts module for latest cpp-sdk and newer protobuf version.
- Adds documentation for Raspberry Pi usage.

### CODE REVIEW

File: csi_camera.cpp [hunk 1]
```diff
@@ -46,11 +46,11 @@ void CSICamera::set_attr(const AttributeMap& attrs, const std::string& name, T C
         auto val = proto->proto_value();
 
         if constexpr (std::is_same<T, int>::value) {
-            this->*member = val.has_number_value() ? val.number_value() : de;
+            this->*member = val.number_value();
```
Comment (seanavery) [inline csi_camera.cpp:49]: Newer `protobuf` version no longer includes `has_X_value` member.

File: csi_camera.cpp [hunk 2]
```diff
@@ -66,7 +66,7 @@ void CSICamera::reconfigure(const Dependencies deps, const ResourceConfig cfg) {
     init(attrs);
 }
 
-Camera::raw_image CSICamera::get_image(const std::string mime_type) {
+Camera::raw_image CSICamera::get_image(const std::string mime_type, const AttributeMap& extra) {
```
Comment (seanavery) [inline csi_camera.cpp:69]: Camera API in cpp-sdk changed here.

File: etc/viam-csi-jetson-arm64.yml [hunk 3]
```diff
@@ -20,10 +20,10 @@ AppDir:
   files:
     include:
     - /usr/local/lib/libviamsdk.so.noabi
+    - /lib/aarch64-linux-gnu/libpthread*
```
Comment (seanavery) [inline etc/viam-csi-jetson-arm64.yml:23]: Had to add `pthread` include explicitly in bundle when moving from focal to jammy.

File: README.md [hunk 4]
```diff
@@ -5,37 +5,44 @@
...
 
-### usage
+You can view more platform specific details at [JETSON.md](./doc/JETSON.md) and [PI.md](./doc/PI.md).
+
+_Note: On a Raspberry Pi, you must install GStreamer plugins before running the module._
 
-1. Download appimage from releases page.
 ```bash
-sudo wget http://packages.viam.com/apps/csi-camera/viam-csi-latest-aarch64.AppImage -O /usr/local/bin/viam-csi
+sudo apt install libgstreamer-plugins-base1.0-dev
```
Comment (bazile-clyde) [inline README.md:17]: If users only need to install `base1` why does our makefile install `good/bad1`? We should probably make our setups and the docs users will use consistent, so we can catch any errors they may have. Here and in PI.md

**Review by bazile-clyde — APPROVED** [review]
Only one comment but otherwise, LGTM! The docs clean up make things a lot more clear btw! Great job!

File: README.md (hunk 4 above)
Comment (seanavery) [reply README.md]: Fixed gstreamer dependency instructions here to include base, good and bad plugin sets.

File: README.md [hunk 5]
```diff
@@ -5,37 +5,46 @@
...
+### Getting Started
 
 For user documentation, see [Add a CSI Camera as a Modular Resource](https://docs.viam.com/extend/modular-resources/examples/csi/).
 
-### usage
+You can view more platform specific details at [JETSON.md](./doc/JETSON.md) and [PI.md](./doc/PI.md).
+
+_Note: On a Raspberry Pi, you must install GStreamer plugins before running the module._
+
+_WARNING: There is a known issue for Debian Bookworm due to changes in the libcamerasrc plugin._
```
Comment (seanavery) [inline README.md:16]: This merge does not include a bundle that works on Bookworm.

**Review by seanavery — COMMENTED** [review]
This merge works on Jetson and Pi Bullseye devices. On Bookworm, there is an issue linking the new libcamera gst plugin which includes trace `ltt` embeds that  deadlock when running inside of an appimage.

**Pull request merged**